### DISCUSSION TITLE
Is index.md the only option?

### DISCUSSION THREAD

**Discussion by andrew-culver** [retypeapp/retype#511] [Q&A]
I would like to have the name of the default file loaded for a subdirectory URL be set to something other than index.md, on a directory-by-directory basis, include the site root. Possible?

**Comment by geoffreymcgill** [comment]
Other options include `default.md` and `readme.md`, see

https://retype.com/configuration/reserved_words/#default-pages

Another option that provides unlimited flexibility is the Page `route` config, see 

https://retype.com/configuration/page/#route

If neither of those are ideal for your requirements, just let us know some more details and we might be able to come up with some other options. 

Hope this helps.

**Comment by andrew-culver** [comment]
Not there yet. 

What I am after is essentially a redirect, so that every load of andrewculver.com/work/ redirects to andrewculver.com/work/introduction/ (introduction.md is a file in the /work/ directory), and have this done for all sub-directories. In other words, set anything to be the default, or than index, readme, and default, in each sub-directory.

Ultimately what I am trying to get away from is having something more informative than an index.md file in every directory, so that search results, editor file lists, file mapping tools (like Obsidian), etc. don't just show "index" all over the place.

**Comment by andrew-culver** [comment]
Here's a thought that might make this easier to implemen: 

The "default" .md file for any directory could be default.md, index.md, readme.md, or, [directory-name].md

andrewculver.com/work/jjRover/jjRover.md is the default for it's directory andrewculver.com/work/jjRover

**Reply by DannyBen** [reply]
Exactly what I was writing while you entered this.

---

Interesting use case. A logical solution to having a "file that describes the directory itself but has a unique name" is to have support for a file that is named exactly like the directory.

- `Installation/Installation.md`
- `Advanced Features/Advanced Features.md`

**Reply by geoffreymcgill** [reply]
Seems there are two scenarios being discussed here. 

1. Redirecting `/work` to `/work/introduction`
2. Custom default page for a directory. 

Scenario 1 is possible with the current release by adding the following `index.md` (or `default.md` or `readme.md`) file inside the `/work` folder:

```md
---
redirect: introduction.md
---
```

Scenario 2 seems like a good idea and I think we should be able to add the functionality into the next release of Retype. 

I think we would treat any combination of a space separator ` `, a dash `-`, and an underscore `_` as equal separators. For example, all of the following would be valid:

- `Advanced Features/Advanced Features.md`
- `Advanced Features/Advanced-Features.md`
- `Advanced Features/Advanced_Features.md`
- `Advanced-Features/Advanced Features.md`
- `Advanced-Features/Advanced-Features.md`
- `Advanced-Features/Advanced_Features.md`
- `Advanced_Features/Advanced Features.md`
- `Advanced_Features/Advanced-Features.md`
- `Advanced_Features/Advanced_Features.md`

**Reply by andrew-culver** [reply]
I'd like to bump this (custom default page for a directory based on the directory name) as and add for Retype 3.0.

**Reply by geoffreymcgill** [reply]
We will try.

**Reply by geoffreymcgill** [reply]
We have added support for the same-name default page functionality. This will enable the `Installation/Installation.md` scenario where `Installation.md` would now be the default page and create the URL `/installation`, instead of the previous `/installation/installation`.

The folder and file name need to match exactly, although case-insensitive matching. We decided to not implement the mixed `-`, `_`, and `<space>` separator support as we felt it would just complicate the scenario, so we're just keeping it simple. 

The functionality will be included in Retype v3.0. There's just one more issue we're dealing with and then will release v3.0.

**Comment by andrew-culver** [comment]
Handling the spaces in the .md file names like this is a great idea.

**Comment by andrew-culver** [comment]
Using the redirect, the left side navigation shows a "Welcome" link/page at the top, even though it is never seen (it redirects as expected). Ideally "Welcome" would not be there.

**Comment by geoffreymcgill** [comment]
Retype requires a root "home page". You can `redirect` that home page to another page, but ultimately there does need to be a root landing page for the project.

Here are a couple ideas to try:

### Option 1

Set `url: andrewculver.com/work` in your `retype.yml`. This will instruct Retype that the public location for the site will be in the `/work` sub-directory. Then set `redirect: introduction.md` in the root `index.md` file.

Here are the three files to demonstrate:
#### retype.yml
```yml
input: .
output: .retype
url: andrewculver.com/work
branding:
  title: Andrew Culver
```
#### index.md
```md
---
redirect: introduction.md
icon: home
title: Home
---
```
#### introduction.md
```md
---
icon: star
---
# Introduction
```

### Option 2

Similar to Option 1, but delete `index.md` and rename `introduction.md` to `index.md`.

#### retype.yml

```yml
input: .
output: .retype
url: andrewculver.com/work
branding:
  title: Andrew Culver
```

#### index.md

```
---
icon: star
---
# Introduction
```

**Reply by geoffreymcgill** [reply]
A third option would be the following, although not as clean as the 1 & 2 above. 

<img width="309" alt="Screen Shot 2023-03-06 at 8 43 20 PM" src="https://user-images.githubusercontent.com/62210/223315364-f2d7e208-01d6-42bc-96c3-f1ee6c591246.png">

#### retype.yml

```yml
input: .
output: .retype
url: andrewculver.com
branding:
  title: Andrew Culver
```

#### index.md

```md
---
redirect: /work/introduction.md
icon: home
title: Home
---
```

#### /work/index.md

```md
---
redirect: introduction.md
title: Work
icon: briefcase
---
```

#### /work/introduction.md

```md
---
icon: star
---
# Introduction
```

<img width="445" alt="Screen Shot 2023-03-06 at 8 40 15 PM" src="https://user-images.githubusercontent.com/62210/223315290-ed7290d2-0a84-4a4f-9829-6c88b041eeee.png">

Hope this helps.
